module_renamer/commands/track_modifications.py

```diff
@@ -0,0 +1,206 @@
+    list_with_modified_imports = {
+        (origin.module + "." + origin.name, working.module + "." + working.name)
+        for origin in origin_filtered
+        for working in working_filtered
+        if working.name is not '*'
+        if origin.name == working.name
+        if origin.module != working.module
+    }
+    return list_with_modified_imports
+
+
+def _check_for_conflicts(list_with_modified_imports):
+    import_from = [modified_import[0] for modified_import in list(list_with_modified_imports)]
+    imports_with_conflict = [class_name
+                             for class_name, number_of_occurrences in Counter(import_from).items()
+                             if number_of_occurrences > 1]
+    if len(imports_with_conflict) > 0:
+        echo(CONFLICT_MSG.format('\n -> '.join(imports_with_conflict)))
+
+        if confirm(INFORMATIVE_CONFLICT_MSG, abort=True):
+            list_with_modified_imports = [modified_import for modified_import in
+                                          list_with_modified_imports
+                                          if modified_import[0] not in imports_with_conflict
+                                          if modified_import[1] not in imports_with_conflict]
+    return list_with_modified_imports
+
+
+def get_imports(project_path, file_counter):
+    # type: (str) -> Import
+    """
+    Look for all .py files on the given project path and return the import statements found on
+    each file.
+
+    Note.: I inserted the TQDM here because was the only way that I could have an accurate
+    progress bar, feel free to share any thoughts or tips on how to improve this progress bar =)
+
+    :type project_path: str
+    :rtype: commands.utils.Import
+    """
+    with tqdm(total=file_counter, unit='files', leave=False, desc=project_path) as pbar:
+        for file_path in walk_on_py_files(project_path):
+            pbar.update()
+            with open(file_path, mode='r') as file:
+                file_content = ast.parse(file.read(), file_path)
+
+            for node in ast.iter_child_nodes(file_content):
+                if isinstance(node, ast.Import):
+                    module = ''
+                elif isinstance(node, ast.ImportFrom):
+                    # node.module can be None on the following situation
+                    #   from . import foo
+                    if node.module is not None:
+                        module = node.module
+                    else:
+                        module = ''
+                else:
+                    continue
+
+                for n in node.names:
+                    yield Import(module, n.name)
+
+
+def walk_on_py_files(folder):
+    """
+    Walk through each python files in a directory
+    """
+    for dir_path, _, files in os.walk(folder):
+        for filename in fnmatch.filter(files, '*.py'):
+            yield os.path.abspath(os.path.join(dir_path, filename))
+
+
+def total_of_py_files_on_project(project_path):
```

Comment:
As I said previously, it seems like a waste to recursively iterate over all files and then throw the result away.

Reply:
- [x] Fixed on commit d6b11ea0a7656b058ce8216ec9387fa30c228830